cpp/src/parquet/column_reader.cc

```diff
@@ -386,6 +395,28 @@ std::shared_ptr<Page> SerializedPageReader::NextPage() {
       throw ParquetException("Invalid page header");
     }
 
+    const PageType::type page_type = LoadEnumSafe(&current_page_header_.type);
+    const bool is_data_page =
+        page_type == PageType::DATA_PAGE || page_type == PageType::DATA_PAGE_V2;
+
+    // Once we have the header, we will call the skip_page_call_back_ to
+    // determine if we should be skipping this page. If yes, we will advance the
+    // stream to the next page.
+    if (skip_page_callback_ && is_data_page) {
+      std::variant<format::DataPageHeader, format::DataPageHeaderV2> data_page_header;
+      if (page_type == PageType::DATA_PAGE) {
+        data_page_header = current_page_header_.data_page_header;
+      } else {
+        data_page_header = current_page_header_.data_page_header_v2;
+      }
+      std::unique_ptr<DataPageStats> data_page_stats =
```

Comment:
one other thing to consider is whther statistics are correct: https://github.com/apache/arrow/blob/147b5c922efe19d34ef7e7cda635b7d8a07be2eb/cpp/src/parquet/metadata.cc#L1325

Reply:
I think the caller should check that the stats are valid before setting a callback function.

Reply:
I made that clear in the comments for setting the callback function.